js/piwik.js

```diff
@@ -2484,6 +2490,66 @@ if (typeof Piwik !== 'object') {
                 }
             }
 
+            /*
+             * Sets up the heart beat timeout.
+             */
+            function heartBeatUp(delay) {
+                if (heartBeatTimeout) {
```

Comment:
We have a few checks re `lastTrackerRequestTime!== null` before calling this method and before calling `setupHeartBeat()`. Can we maybe move this check into here to make sure we never send any heart beat when there is not a tracking request sent yet. This will make calling this method easier and will prevent possible problems in the future but I get why it is not done here. I'd personally move this check into here but not mandatory.